aom/src/main/java/com/nedap/archie/aom/primitives/CTerminologyCode.java

```diff
@@ -63,17 +66,17 @@ public boolean isValidValue(TerminologyCode value) {
         if(getConstraint().isEmpty()) {
             return true;
         }
-        for(String constraint:getConstraint()) {
-            if(constraint.startsWith("at")) {
-                if(value.getCodeString() != null && value.getCodeString().equals(constraint)) {
-                    return true;
-                }
-            } else if (constraint.startsWith("ac")) {
-                if(value.getTerminologyId() != null && value.getTerminologyId().equals(constraint)) {
-                    return true;
-                }
-            }
+        if(value != null && value.getTerminologyId() != null && !value.getTerminologyId().equalsIgnoreCase("local") && !AOMUtils.isValueSetCode(value.getTerminologyId())) {
+            //this is a non-local terminology. If a term binding is there, we may be able to validate, if external, we wil not be able to
+            //TODO: check term bindings as well! However, no idea how to handle the URIs, as the value will not be URI. I think?
+            //so return true for now for non-local terminology values
+            return true;
         }
+        List<String> values = this.getValueSetExpanded();
```

Comment:
This doesn't seem to work in Operational Templates when the rules come from an inner archetype. It tries to get the valueset from the wrong archetype.

Reply:
Thanks, that should be fixable, I'll fix that!

Reply:
Turns out, this is pretty hard to fix:
- we flatten the rules into an OPT
- the rules are all at OPT level
- we don't know which rules come from which part of the archetype.
- that makes it hard to determine the correct place to get the ac-code from.

possible solutions: 
- find the nearest for all loop, and use that archetype as context?

Going to try that...

Reply:
I did that. it does mean this works:
`/some/path/to/term/code matches {[ac3]}`

But if not a path but an expression that comes from several locations in the archetype, it does not. Not fixable except by doing drastic changes, by adding extra fields to the OPT model.